response_operations_ui/controllers/party_controller.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+import requests
+from structlog import wrap_logger
+
+from response_operations_ui import app
+from response_operations_ui.exceptions.exceptions import ApiError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_party_by_ru_ref(ru_ref):
+    logger.debug('Retrieving reporting unit', ru_ref=ru_ref)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/parties/type/B/ref/{ru_ref}'
+    response = requests.get(url, auth=app.config['PARTY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.error('Failed to retrieve reporting unit', ru_ref=ru_ref)
+        raise ApiError(response)
+
+    logger.debug('Successfully retrieved reporting unit', ru_ref=ru_ref)
+    return response.json()
+
+
+def get_business_party_by_party_id(business_party_id, collection_exercise_id=None):
+    logger.debug('Retrieving business party',
+                 business_party_id=business_party_id, collection_exercise_id=collection_exercise_id)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/businesses/id/{business_party_id}'
+    params = {"collection_exercise_id": collection_exercise_id, "verbose": True}
+    response = requests.get(url, params=params, auth=app.config['PARTY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.error('Error retrieving business party',
+                     business_party_id=business_party_id, collection_exercise_id=collection_exercise_id)
+        raise ApiError(response)
+
+    logger.debug('Successfully retrieved business party',
+                 business_party_id=business_party_id, collection_exercise_id=collection_exercise_id)
+    return response.json()
+
+
+def get_respondent_party_by_party_id(respondent_party_id):
+    logger.debug('Retrieving respondent party', respondent_party_id=respondent_party_id)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/respondents/id/{respondent_party_id}'
+    response = requests.get(url, auth=app.config['PARTY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.error('Error retrieving respondent party', respondent_party_id=respondent_party_id)
```

Comment:
The get respondent api can respond with a 400 or 404 which we don't want to log at error level.

```
log_level = logger.warning if response.status_code in (400, 404) else logger.exception
log_level("Respondent retrieval failed", respondent_party_id=respondent_party_id)
```